practice_app/pages/api/getAllProducts.js

```diff
@@ -0,0 +1,46 @@
+const getAllProducts = (req, res) =>
+  new Promise(async (resolve) => {
+    // Imports the Google Cloud client library
```

Comment:
It is enough to do `const getAllProduct = async (req, res) =>` without any explicit Promise as an `async` function is a function that returns a `Promise itself`. Also remove any `res.on('finish', resolve)` as it is not needed and simply use the regular `res.json({...})`